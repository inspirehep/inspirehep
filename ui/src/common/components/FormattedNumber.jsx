import PropTypes from 'prop-types';
import { addCommasToNumber } from '../utils';

function FormattedNumber({ children }) {
  return addCommasToNumber(children);
}

FormattedNumber.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
};

export default FormattedNumber;
