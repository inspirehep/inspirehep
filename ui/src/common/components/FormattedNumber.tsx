import React from 'react';

import { addCommasToNumber } from '../utils';

interface FormattedNumberProps {
  children: string | number;
}

function FormattedNumber({ children }: FormattedNumberProps) {
  return <>{addCommasToNumber(children)}</>;
}

export default FormattedNumber;
