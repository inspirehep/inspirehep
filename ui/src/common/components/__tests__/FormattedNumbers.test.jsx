import React from 'react';
import { shallow } from 'enzyme';
import FormattedNumber from '../FormattedNumber';

describe('FormattedNumber', () => {
  it('renders with children', () => {
    const wrapper = shallow(<FormattedNumber>{1243553}</FormattedNumber>);
    expect(wrapper).toMatchSnapshot();
  });
});
