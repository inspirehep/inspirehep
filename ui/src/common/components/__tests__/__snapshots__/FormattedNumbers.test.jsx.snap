// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedNumber renders with children 1`] = `"1,243,553"`;
